shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -939,6 +942,117 @@ public void setLocalizationPlugin(@Nullable LocalizationPlugin localizationPlugi
 
   // ----- End Localization Support ----
 
+  // ----- Start Dynamic Features Support ----
+
+  /** Sets the dynamic feature manager that is used to download and install split features. */
+  @UiThread
+  public void setDynamicFeatureManager(@NonNull DynamicFeatureManager dynamicFeatureManager) {
+    ensureRunningOnMainThread();
+    this.dynamicFeatureManager = dynamicFeatureManager;
+  }
+
+  /**
+   * Called by dart's deferred load handler to request that a dart deferred library corresponding
+   * to loadingUnitId be downloaded (if necessary) and loaded into the dart vm.
+   *
+   * <p>This method delegates the task to DynamicFeatureManager, which handles the download and
+   * loading of the dart library and any assets.
+   *
+   * @param loadingUnitId The loadingUnitId is assigned during compile time by gen_snapshot and is
+   *                      automatically retrieved when loadLibrary() is called on a dart deferred
+   *                      library.
+   */
+  @SuppressWarnings("unused")
+  @UiThread
+  public void RequestDartDeferredLibrary(int loadingUnitId) {
+    dynamicFeatureManager.downloadDynamicFeature(loadingUnitId, null);
+  }
+
+  /**
+   * Searches each of the provided paths for a valid dart shared library .so file and resolves
+   * symbols to load into the dart VM.
+   *
+   * <p>Successful loading of the dart library completes the future returned by loadLibrary() that
+   * triggered the install/load process.
+   *
+   * @param loadingUnitId The loadingUnitId is assigned during compile time by gen_snapshot and is
+   *                      automatically retrieved when loadLibrary() is called on a dart deferred
+   *                      library. This is used to identify which dart deferred library the resolved
+   *                      correspond to.
+   *
+   * @param searchPaths   An array of paths in which to look for valid dart shared libraries. This
+   *                      supports paths within zipped apks as long as the apks are not compressed
+   *                      using the `path/to/apk.apk!path/inside/apk/lib.so` format. Paths will be
+   *                      tried first to last and ends when a library is sucessfully found. When the
+   *                      found library is invalid, no additional paths will be attempted.
+   */
+  @UiThread
+  public void loadDartDeferredLibrary(
+      int loadingUnitId,
+      @NonNull String[] searchPaths) {
+    ensureRunningOnMainThread();
+    ensureAttachedToNative();
+    nativeLoadDartDeferredLibrary(nativePlatformViewId, loadingUnitId, searchPaths);
+  }
+
+  private native void nativeLoadDartDeferredLibrary(
+      long nativePlatformViewId,
```

Comment:
nativePlatformViewId shouldn't be here right?

Reply:
We use the nativePlatformViewId to access the platform view which is our entry point into the engine